chalk-solve/src/coherence.rs

```diff
@@ -16,6 +17,7 @@ where
     DB: ChalkSolveDatabase,
 {
     db: &'db DB,
+    solver_choice: SolverChoice,
```

Comment:
Similar to my comment in `wf.rs`, I think we could store the solver *here* (and instantiate it in the `new` method) instead of the `solver_choice`. But this is purely an efficiency thing.